src/feature/sift.cc

```diff
@@ -206,22 +206,24 @@ void FindBestMatchesOneWayFLANN(
         indices,
     Eigen::Matrix<int, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>*
         distances) {
+  if (query.rows() == 0 || database.rows() == 0) {
+    return;
+  }
+
   const size_t kNumNearestNeighbors = 2;
   const size_t kNumTreesInForest = 4;
 
-  indices->resize(query.rows(), std::min(kNumNearestNeighbors,
-                                         static_cast<size_t>(database.rows())));
-  distances->resize(
-      query.rows(),
-      std::min(kNumNearestNeighbors, static_cast<size_t>(database.rows())));
+  if (kNumNearestNeighbors > database.rows()) {
```

Comment:
It's not really relevant for the COLMAP codebase, but technically we would still populate the indices/distances output and just skip the FLANN code block below.